nixos/modules/services/hardware/tlp.nix

```diff
@@ -45,77 +34,37 @@ in
         default = "";
         description = "Additional configuration variables for TLP";
       };
-
     };
-
   };
 
-
   ###### implementation
-
   config = mkIf cfg.enable {
+    boot.kernelModules = [ "msr" ];
 
-    powerManagement.scsiLinkPolicy = null;
-    powerManagement.cpuFreqGovernor = null;
-    powerManagement.cpufreq.max = null;
-    powerManagement.cpufreq.min = null;
-
-    systemd.sockets.systemd-rfkill.enable = false;
-
-    systemd.services = {
-      "systemd-rfkill@".enable = false;
-      systemd-rfkill.enable = false;
-
-      tlp = {
-        description = "TLP system startup/shutdown";
-
-        after = [ "multi-user.target" ];
-        wantedBy = [ "multi-user.target" ];
-        before = [ "shutdown.target" ];
-        restartTriggers = [ confFile ];
-
-        serviceConfig = {
-          Type = "oneshot";
-          RemainAfterExit = true;
-          ExecStart = "${tlp}/bin/tlp init start";
-          ExecStop = "${tlp}/bin/tlp init stop";
-        };
-      };
-
-      tlp-sleep = {
-        description = "TLP suspend/resume";
-
-        wantedBy = [ "sleep.target" ];
-        before = [ "sleep.target" ];
+    environment.etc = {
+      "tlp.conf".source = extraConfig;
```

Comment:
I believe you want this from `cfg.extraConfig`? eval failed

Reply:
Correct, fixed :)